From c7bdfa014bac092cc4490820708ad65da07cf15c Mon Sep 17 00:00:00 2001
From: Hans Verkuil <hans.verkuil@cisco.com>
Date: Fri, 7 Jun 2013 10:22:51 +0200
Subject: [PATCH 5/6] rds-ctl: support -d10 to refer to radio10.

In order to be consistent with other v4l utilities support the '-d<num>' option
for num >= 10.

Signed-off-by: Hans Verkuil <hans.verkuil@cisco.com>
---
 utils/rds-ctl/rds-ctl.cpp | 9 +++++----
 1 file changed, 5 insertions(+), 4 deletions(-)

diff --git a/utils/rds-ctl/rds-ctl.cpp b/utils/rds-ctl/rds-ctl.cpp
index d894ec1..a9fe2a8 100644
--- a/utils/rds-ctl/rds-ctl.cpp
+++ b/utils/rds-ctl/rds-ctl.cpp
@@ -133,7 +133,7 @@ static void usage_common(void)
 	       "  --all              display all device information available\n"
 	       "  -D, --info         show driver info [VIDIOC_QUERYCAP]\n"
 	       "  -d, --device=<dev> use device <dev>\n"
-	       "                     if <dev> is a single digit, then /dev/radio<dev> is used\n"
+	       "                     If <dev> starts with a digit, then /dev/radio<dev> is used\n"
 	       "                     default: checks for RDS-capable devices,\n"
 	       "                     uses device with lowest ID\n"
 	       "  -h, --help         display this help message\n"
@@ -763,9 +763,10 @@ static int parse_cl(int argc, char **argv)
 		switch (opt) {
 		case OptSetDevice:
 			strncpy(params.fd_name, optarg, 80);
-			if (isdigit(optarg[0]) && optarg[1] == 0) {
-				char newdev[20];
-				sprintf(newdev, "/dev/radio%c", optarg[0]);
+			if (optarg[0] >= '0' && optarg[0] <= '9' && strlen(optarg) <= 3) {
+				static char newdev[20];
+
+				sprintf(newdev, "/dev/radio%s", optarg);
 				strncpy(params.fd_name, newdev, 20);
 			}
 			break;
-- 
1.8.2.1

